libs/langchain/langchain/storage/upstash_redis.py

```diff
@@ -117,3 +118,57 @@ def yield_keys(self, *, prefix: Optional[str] = None) -> Iterator[str]:
                     yield relative_key
                 else:
                     yield key
+
+
+@deprecated("0.0.335", alternative="UpstashRedisByteStore")
+class UpstashRedisStore(_UpstashRedisStore):
+    """
+    BaseStore implementation using Upstash Redis
+    as the underlying store to store strings.
+
+    Deprecated in favor of the more generic UpstashRedisByteStore.
+    """
+
+
+class UpstashRedisByteStore(BaseStore[str, bytes]):
+    """
+    BaseStore implementation using Upstash Redis
+    as the underlying store to store raw bytes.
+    """
+
+    def __init__(
+        self,
+        *,
+        client: Any = None,
+        url: Optional[str] = None,
+        token: Optional[str] = None,
+        ttl: Optional[int] = None,
+        namespace: Optional[str] = None,
+    ) -> None:
+        self.underlying_store = _UpstashRedisStore(
+            client=client, url=url, token=token, ttl=ttl, namespace=namespace
+        )
+
+    def mget(self, keys: Sequence[str]) -> List[Optional[bytes]]:
+        """Get the values associated with the given keys."""
+        return [
+            value.encode("utf-8") if value is not None else None
```

Comment:
I can't figure out how this ends up being a string. Redis works with bytes not strings... is upstash redis doing some additional encoding/decoding?

Reply:
I guess so? The underlying implementation returns a string 🤷